module/setup.go

```diff
@@ -6,20 +6,58 @@ import (
 
 	clienttypes "github.com/cosmos/ibc-go/v8/modules/core/02-client/types"
 	"github.com/cosmos/ibc-go/v8/modules/core/exported"
-	"github.com/datachainlab/ibc-parlia-relay/module/constant"
 	"github.com/hyperledger-labs/yui-relayer/core"
 	"github.com/hyperledger-labs/yui-relayer/log"
 )
 
+const skip = 100
+
 type queryVerifiableNeighboringEpochHeaderFn = func(context.Context, uint64, uint64) (core.Header, error)
 
+func shouldSubmitBoundaryTimestampHeader(
+	getHeader getHeaderFn,
+	trustedBlockNumber uint64,
+	latestFinalizedBlockNumber uint64,
+	forkSpecs []*ForkSpec) (*uint64, uint64, error) {
+
+	trustedBlock, err := getHeader(context.Background(), trustedBlockNumber)
+	if err != nil {
+		return nil, 0, err
+	}
+	latestFinalizedBlock, err := getHeader(context.Background(), latestFinalizedBlockNumber)
+	if err != nil {
+		return nil, 0, err
+	}
+
+	latestForkSpec := forkSpecs[len(forkSpecs)-1]
+	latestCondition := latestForkSpec.GetHeightOrTimestamp()
+	if x, ok := latestCondition.(*ForkSpec_Timestamp); ok {
+		if MilliTimestamp(trustedBlock) < x.Timestamp && x.Timestamp < MilliTimestamp(latestFinalizedBlock) {
+			boundaryHeight, err := GetBoundaryHeight(getHeader, latestFinalizedBlock.Number.Uint64(), *latestForkSpec)
+			if err != nil {
+				return nil, 0, err
+			}
+			// Must be right before boundary height
+			if boundaryHeight.Height == 0 {
+				return nil, 0, fmt.Errorf("boundary height not found")
+			}
+
+			nextForkBoundaryHeightMinus1 := uint64(boundaryHeight.Height) - 1
+			log.GetLogger().Info("ForkSpec height required", "ts", x.Timestamp, "height", boundaryHeight, "nextForkBoundaryHeightMinus1", nextForkBoundaryHeightMinus1)
```

Comment:
The height corresponding to the HF timestamp can be immediately checked in this log.